scripts/install-chef-suse.sh

```diff
@@ -665,7 +671,25 @@ service chef-client status &> /dev/null && service chef-client stop
 if ! [ -e ~/.chef/knife.rb -a -e ~/.chef/root.pem ]; then
     # no_proxy is currently not supported in ruby see bsc#958716
     # unset it for this call
-    (unset http_proxy; yes '' | knife configure -i)
+    (unset http_proxy
+    if [ knife client list | grep -q root ]; then
+        yes "Y" | knife client delete root
+    fi
+    /usr/bin/expect <<EOD
+spawn knife configure -i
+expect {
+  "Overwrite /root/.chef/knife.rb? (Y/N)" { send "Y\r"; exp_continue }
+  "Where should I put the config file? *" { send "/root/.chef/knife.rb\r"; exp_continue }
+  "Please enter the chef server URL: *" { send "http://localhost:4000\r"; exp_continue }
+  "Please enter a clientname for the new client: *" { send "root\r"; exp_continue }
+  "Please enter the existing admin clientname: *" { send "chef-webui\r"; exp_continue }
+  "Please enter the location of the existing admin client's private key: *" { send "/etc/chef/webui.pem\r"; exp_continue }
+  "Please enter the validation clientname: *" { send "chef-validator\r"; exp_continue }
+  "Please enter the location of the validation key: *" { send "/etc/chef/validation.pem\r"; exp_continue }
+  "Please enter the path to a chef repository (or leave blank):" { send "\r"; exp_continue }
+}
+EOD
```

Comment:
I'm also wondering if we really need expect here; did you try simply using parameters for knife configure? Something like `knife configure -i -u root --defaults --repository ""`